cmd/crossplane/core/core.go

```diff
@@ -59,7 +69,35 @@ func (c *Command) Run(log logging.Logger) error {
 		return errors.Wrap(err, "Cannot get config")
 	}
 
-	mgr, err := ctrl.NewManager(cfg, ctrl.Options{SyncPeriod: &c.Sync})
+	// approach 1:
+	// abstract away `ctrl.NewManager` call in crossplane-runtime
+	// ----------------------------------------------------------
+	// mgr, err := controller.NewManager(cfg, controller.Options{
+	// 	Options: &ctrl.Options{
+	// 		LeaderElection:          c.LeaderElection,
+	// 		LeaderElectionNamespace: c.Namespace,
+	// 		SyncPeriod:              &c.Sync,
+	// 	},
+	// 	ManagerName:      c.Name,
+	// 	ManagerNamespace: c.Namespace,
+	// })
+
+	// approach 2:
+	// keep calling `ctrl.NewManager` with additional config here
+	// ----------------------------------------------------------
+	if c.Name == "" {
+		return errors.New(errManagerName)
+	}
+	if c.Namespace == "" {
+		return errors.New(errManagerNamespace)
+	}
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		LeaderElection:             c.LeaderElection,
+		LeaderElectionID:           fmt.Sprintf("crossplane-leader-election-%s", c.Name),
+		LeaderElectionNamespace:    c.Namespace,
```

Comment:
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.6.3/pkg/leaderelection/leader_election.go#L61

It's not documented very well, but it seems like if we omit this we'll use the namespace where the pod is running by default. I think it might be simpler overall to just do that, so I'd prefer to not set `LeaderElectionNamespace` unless we get requests for it to be configurable.

Reply:
That's true, it might have been slight over-engineering on my side to expose the ability to provide any arbitrary namespace for leader election resource, which in reality there might never a case for this separation. And if so it can easily be added back again.

Reply:
Actually, I think we should revert this back, without `LeaderElectionNamespace: c.Namespace` there won't be any way to manually run this out of a cluster (i.e. `go run cmd/crossplane/main.go core ...` won't work) with following error:

```
main: error: cannot run crossplane: Cannot create manager: unable to find leader election namespace: not running in-cluster, please specify LeaderElectionNamespace
```

Reply:
I wonder how likely it is that folks will want to run Crossplane out of cluster _and_ also need leader election? For the `go run` use case it seems like we'd be developing Crossplane, and thus probably would not need leader election enabled.

Reply:
It is most probably very less likely, and for the latter you are correct, us running `go run` we won't be needing leader election but without that extra attribute for `ctrl.Options` it will never work.

Reply:
In 38d7b36 I effectively reversed the behavior, it means leader election is disabled by default (`go run` continues to work as before and still can be enabled with `--leader-election` flag or `LEADER_ELECTION=true` environment variable), and in the helm chart it gets enabled by default upon deployment.